### PR TITLE
Documentation for pybind_wrap function

### PR DESCRIPTION
**Pull request by varunagrawal**
Added documentation for all the arguments in `pybind_wrap`.

### CODE REVIEW

File: cmake/PybindWrap.cmake [hunk 1]
```diff
@@ -27,15 +27,23 @@ set(PYBIND11_PYTHON_VERSION ${WRAP_PYTHON_VERSION})
...
-# install_path: destination to install the library libs: libraries to link with
-# dependencies: Dependencies which need to be built before the wrapper
+# generated_cpp: The name of the cpp file which is generated from the tpl file.
+# module_name: The name of the Python module to use.
+# top_namespace: The C++ namespace under which the code to be wrapped exists.
+# ignore_classes: CMake list of classes to ignore from wrapping.
+# install_path: Destination to install the library.
+# module_template: The template file (.tpl) from which to generate the Pybind11 module.
+# libs: Libraries to link with.
+# dependencies: Dependencies which need to be built before the wrapper.
```
Comment (ProfFan) [inline cmake/PybindWrap.cmake:46]: There is one extra arg `use_boost`

**Review by ProfFan — CHANGES_REQUESTED** [review]

File: cmake/PybindWrap.cmake (hunk 1 above)
Comment (varunagrawal) [reply cmake/PybindWrap.cmake:46]: I raised an issue saying that `use_boost` should be `ON` by default. #4

Comment (ProfFan) [reply cmake/PybindWrap.cmake:46]: I think just add it here in the doc first?

**Comment by varunagrawal** [comment]
@ProfFan all done

**Review by ProfFan — APPROVED** [review]
Cool!

**Comment by varunagrawal** [comment]
I can't merge. Can you please grant access to the phd students group?

**Comment by ProfFan** [comment]
@varunagrawal done :)

**Pull request merged**